internal/netTestutils.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
nit: this filename seems a bit...weird. `net_test_utils.go`?  Or maybe `net_utils_test.go` so it only gets compiled into the test binary and not by default?